### DISCUSSION TITLE
Few facts about "compressed" plot.

### DISCUSSION THREAD

**Discussion by smartbitcoin** [smartbitcoin/arc_plot#19] [Announcements]
As  NOSSD and Madmax market their new pool and plot tech as "compressed" plot. There are few things the community must know.

1. **"compressed plot" is marketing term, actually they just skipping store data in plot file and calculate them in GPU on the fly by harvester.**.  
there are nothing about new compressed technology in their product. What those new plotter do is just skipping store table data in plot file ( such as  skip table 1,2,3 and mark it as compress level 1-9 ) and move those data into harvester and calculate them on the fly by GPU.

2. If using those plot format,  chia is not pure POST( prove space and time) anymore.  **Chia is just a hybrid of prove space and time and partial prove of work.**

3. Every time the GPU energy efficiency improved, the sweet point of "skipped storage ( or compressed level )" will change and the motivation to re-plotting will increase. in theory,  whole plot file can be calculated in GPU and no prove of space required by chia anymore. in reality, with rtx 3060, table 2 can be efficiently skipped now. and for rtx 4090 table 3 etc. And every generation of GPU the sweet point of POST and POW will shift. It's possible an ASIC hardware shows in market to accelerate this process. so **plotting NEVER end**.

4. It's a **negative-sum  game** for all farmer. in case everybody take the "skipping storage" plot format and by the end , farmer just pay extra GPU harvester hardware and hydro cost and higher dev and pool fees without any gains.

5. I highly suggest chia upgrade to at least K36 if we wanna keep this blockchain pure Prove of space and time.

6. arc_plot will release those type of hybrid plotter if have enough resource and **keep it as freeware for ever.**

**Comment by madMAx43v3r** [comment]
The compute needed to compress more is exponential, so there is a limit, there wont be a POW or replotting armageddon. maybe in 5 years C9 might become profitable, but that's about it. and you fail to consider that storage will also get cheaper over time.

> If using those plot format, chia is not pure POST( prove space and time) anymore.

Correct, it will be 5-10% PoW, is that so bad?

> I highly suggest chia upgrade to at least K36 if we wanna keep this blockchain pure Prove of space and time.

That would be insane. In any case, I already have a GPU cluster plotter than can plot any K size fully in RAM, which obviously only the biggest whales would be able to afford. GPU farming compressed k36+ is also possible, in fact a GPU would be required even for the lower levels of compressed k36+ plots. It would only make it worse and give whales a much bigger advantage.

**Reply by madMAx43v3r** [reply]
That's why in MMX I chose the min K size to be k30, so you can GPU plot fully in RAM with just 64 GB. Most gaming rigs will be able to do that, no server hardware needed.

**Reply by smartbitcoin** [reply]
> Correct, it will be 5-10% PoW, is that so bad?

PoST or PoW as consensus algorithm, can't say which one is good or bad.  but chia pick PoST as one of it's feature to be BTC 2.0. If it go hybird of PoST and PoW,  my concern is the chia IPO maybe failed as the green ethos was broken.

>  I already have a GPU cluster plotter than can plot any K size fully in RAM

The point was not about GPU can brutal force chia plot or not. It's about "the sweet point" of efficiency GPU on the fly plots vs the storage persistent plots. Now rtx 40xx serious have better efficiency ( or monetary incentives  ) in table 2 and 3, which is max to 30-40% of storage space. Farmer only do PoST will lose to hybrid farming. Increase K value can shift the sweet point back to persistancy storage farmer and keep chia as PoST 's green logo.

There few other solution instead of just increase K ,  like add table more than 7,  or change the way Table 1 generated by add more memory heavy operation ( like traditional anti -asic way ) etc , but increase K is still the simplest and more compatible way to achieve that. 

In IC industry , the logic unit ( like cpu ,gpu ) vs the storage unit ( like ram, disk ),  the previous have 2-4x higher growth rate vs later one.  so if the barrier is not high enough,  the sweet point between PoW and PoST alwasy shift to PoW side, so full ASIC chia farming solution will show in market especially when next crypto bull market coming push chia to higher price.

**Reply by madMAx43v3r** [reply]
I don't think you understand exponential functions. C10 is essentially a full phase 1 table 1 + 2 + 3 + 4 recompute to get a full proof. It's the end of the road, there is nothing to compress the plots further at the cost of more compute.

I'm trying to say there wont be any major shift towards more PoW, with ASIC or not.

**Reply by smartbitcoin** [reply]
As I am the first public GPU plotter author, there are maybe nobody better than me understood all details. that's why I have to speak out the facts.
What your said maybe it's the end of road of your gigahorse design and maybe part of community like those changes as they own the capability to take advantage of it.  But it's not end of road of GPU or Asic on fly plotting. If you want to say is  b/c gigahorse won't take too much for GPU so this hybrid pow+post will stop at eg 30% of Pow,  don't means this is fare for farmer without GPU capability and don't want do hybrid farming. And still can't stop some type of ASIC also.  It is exponential increase after table 4, that's right , but GPU cluster can do all table in ram and the sweet point between POW and POST for current generation GPU stop at table3 doesn't means it will stop at table 3 for ever.

**Reply by madMAx43v3r** [reply]
You are talking about the re-plotting attack, which has nothing to do with plot compression. It's possible to do the attack today, however it's about 10x more expensive than farming with HDDs.
And please consider that HDD cost and power draw will go down as well over time, not just GPU cost and power efficiency.

**Reply by smartbitcoin** [reply]
I am talking about sweet point between POW and POST even after t4 the exponential increase complexity.  Full in ram on the fly plotting is just extreme  example. My testing shows there was a way to efficiently do all t1 to 4 table ( you maybe define it as C10 ) on current generation nvidia rtx 3090 event not 5nm rtx 40 series. If you think your program is the end of road and there was 5 year peace doesn't matter how the IC industry goes and how hellman attack suppose works will be wrong. something you don't realize now doesn't means it not exist, sooner or later people will discover it.

**Comment by smartbitcoin** [comment]
@madMAx43v3r , Btw,  I wrote arc_plot just as a hobby by spare time.  there are no intention to compete with your gigahorse project for dev fees.  As a freeware , if somebody use it , it already my best returns.

**Reply by smartbitcoin** [reply]
@madMAx43v3r   One suggestion for your gigahorse project is:    instead of do GPU on the fly on client harvester side,   it's better build your own pool,  consolidate all GPU calculation on pool side,  and the fee structure is based on client's C level to cover the cost of GPU's expense. this all-include turn key project will benefit small farmer from complicated farming setup.

**Reply by madMAx43v3r** [reply]
I did think about that, however it's too centralized and I don't want to run a pool (that's a whole project in itself).
Having people pay for a service so they can farm is a lot of friction also.

**Reply by digitalspaceport** [reply]
@smartbitcoin If your GPU plotter is a hobby done in spare time, why not fully open source the code and that way it can be developed further by more then just you? You had mentioned in the past that you would. Seems like right now is a good time. I personally have serious reservations about closed source "freeware" that does not have an explicit profit motivation in Cryptocurrency.

**Reply by smartbitcoin** [reply]
@digitalspaceport ,  If you want to deep knowledge about the difference between freeware and open-source,  you can read GNU founder Richard Stallman's article:

https://www.gnu.org/philosophy/open-source-misses-the-point.html

And I share your feeling s in same way, if a commercial or no-technology person asking open source a project, I also highly doubt their motivation explicitly in cyptos. 

The spirit of freedom in software ( freeware ) is: the author have freedom write what they feel interest without legal concerns and the user have freedom to run what ever they feel useful without fear of copyright.  Free software foundation was created if not for show you their code or give your free burger, it's to protect the right of freedom to express their idea in code  or use any software without any patent or copyright issue.

If you have "serious reservations about closed source "freeware"" ,  lots of people also have "serious concerns about crypto as it's for money laundry. "  .  I wish this analogy can make you feel better about freeware and bring more love to crytos.

**Reply by smartbitcoin** [reply]
@digitalspaceport 

I once have plan to open source it ,  but I also need make sure nobody will take the copy of code to attach chia's Prove Of Time and Space. if it's inevitable , at least won't accelerate this process.  Both "compressed plot" from Nossd or Madmax, it's just a variants of Hellman's attack ( instead of reverse table function f1, f2, f3, it shrink x1,x2 of table 1,2,3 and recalculate f1,f2,f3 in gpu on the fly ) to chia network. Somebody like it , the others not, and majority don't even know what it means.  If you want to know more about Hellman's attack in chia architecture,  please read this:

https://github.com/Chia-Network/chiapos/blob/main/documents/proof_of_space.html

section: Hellman-Attacks

**Reply by digitalspaceport** [reply]
> @digitalspaceport , If you want to deep knowledge about the difference between freeware and open-source, you can read GNU founder Richard Stallman's article:
> 
> https://www.gnu.org/philosophy/open-source-misses-the-point.html
> 
> And I share your feeling s in same way, if a commercial or no-technology person asking open source a project, I also highly doubt their motivation explicitly in cyptos.
> 
> The spirit of freedom in software ( freeware ) is: the author have freedom write what they feel interest without legal concerns and the user have freedom to run what ever they feel useful without fear of copyright. Free software foundation was created if not for show you their code or give your free burger, it's to protect the right of freedom to express their idea in code or use any software without any patent or copyright issue.
> 
> If you have "serious reservations about closed source "freeware"" , lots of people also have "serious concerns about crypto as it's for money laundry. " . I wish this analogy can make you feel better about freeware and bring more love to crytos.

I bring it up as you mentioned it yourself.... this does not make me feel better tbh. Profit as a motivator is rational and makes sense from a product standpoint + enriching the developer to incentivize them.

**Reply by smartbitcoin** [reply]
ok,  I'll stop update this project till 100xch donation collected .  does that make more sense for you ? :-)
